spdmlib/src/message/measurement.rs

```diff
@@ -98,7 +98,15 @@ pub struct SpdmMeasurementsResponsePayload {
 
 impl SpdmCodec for SpdmMeasurementsResponsePayload {
     fn spdm_encode(&self, context: &mut common::SpdmContext, bytes: &mut Writer) {
-        self.number_of_measurement.encode(bytes); // param1
+        //When Param2 in the requested measurement operation is 0 , this
+        //parameter shall return the total number of measurement indices on
+        //the device. Otherwise, this field is reserved.
+        if self.measurement_record.number_of_blocks == 0 {
+            // SpdmMeasurementOperation::SpdmMeasurementQueryTotalNumber
+            5_u8.encode(bytes); // param1
```

Comment:
It is still wrong.
See my previous feedback.

Reply:
Got it, will re-submit. Thank you!